Source/WebCore/html/HTMLImageElement.cpp

```diff
@@ -430,6 +434,11 @@ void HTMLImageElement::attributeChanged(const QualifiedName& name, const AtomStr
         m_hadNameBeforeAttributeChanged = willHaveName;
         break;
     }
+#if ENABLE(SPATIAL_IMAGE_CONTROLS)
+    case AttributeNames::controlsAttr:
+        SpatialImageControls::updateSpatialImageControls(*this);
```

Comment:
We need to do this when the image gets updated as well but that could happen in a follow-up.

Reply:
Isn't it sufficient that `updateSpatialImageControls` also gets called by `ImageLoader::notifyFinished` for image updates?

Reply:
Ah, I missed that. Yeah, that'll work.